### PR TITLE
Health Probe fixes for TLS

### PR DESCRIPTION
**Pull request by philbrookes**
Fix health probes for TLS endpoints and allow ignoring bad certificates.

### CODE REVIEW

File: pkg/controllers/dnshealthcheckprobe/dnshealthcheckprobe_controller.go [hunk 1]
```diff
@@ -87,6 +87,7 @@ func (r *DNSHealthCheckProbeReconciler) Reconcile(ctx context.Context, req ctrl.
 			p.Protocol = protocol
 			p.AdditionalHeaders = previous.Spec.AdditionalHeaders
 			p.ExpectedReponses = previous.Spec.ExpectedReponses
+			p.AllowInsecureCertificate = previous.Spec.AllowInsecureCertificate
```
Comment (sergioifg94) [inline pkg/controllers/dnshealthcheckprobe/dnshealthcheckprobe_controller.go:90]: This is setting the flag only when an existing probe is updated, but not when it's newly created. I'd add this also to the `else` block below, line 103

File: pkg/health/queuedProbeWorker.go [hunk 2]
```diff
@@ -181,3 +187,27 @@ func checkResponse(response int, expected []int) bool {
 	}
 	return false
 }
+
+// TransportWithDNSResponse creates a new transport which overrides hostnames.
+func TransportWithDNSResponse(overrides map[string]string) http.RoundTripper {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	dialer := &net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
```
Comment (sergioifg94) [inline pkg/health/queuedProbeWorker.go:196]: I wonder if it makes sense to have these as part of the spec of the health checks. Not necessarily for this PR but might be interesting

**Review by sergioifg94 — COMMENTED** [review]
Mostly looks good, just the comments below

File: pkg/health/queuedProbeWorker.go (hunk 2 above)
Comment (philbrookes) [reply pkg/health/queuedProbeWorker.go:196]: yeah, that's an interesting idea. Let's track that in a Jira?

File: pkg/controllers/dnshealthcheckprobe/dnshealthcheckprobe_controller.go (hunk 1 above)
Comment (philbrookes) [reply pkg/controllers/dnshealthcheckprobe/dnshealthcheckprobe_controller.go:90]: nice, thanks, added now.

File: pkg/health/queuedProbeWorker.go (hunk 2 above)
Comment (sergioifg94) [reply pkg/health/queuedProbeWorker.go:196]: Sounds good, will do. You mean Jira or a GH issue?

**Pull request merged**